src/matcha_ml/runners/azure_runner.py

```diff
@@ -42,6 +45,38 @@ def _show_terraform_outputs(self, matcha_state: MatchaState) -> None:
         resources_json = dict_to_json(resources_dict)
         print_json(resources_json)
 
+    def is_local_state_stale(self) -> bool:
+        """Checks for congruence between the local config file and the local tfvars file."""
+        local_tfvars_file = os.path.join(
+            os.getcwd(),
+            ".matcha",
+            "infrastructure",
+            "remote_state_storage",
+            "terraform.tfvars.json",
+        )
+        local_config_file = os.path.join(os.getcwd(), "matcha.config.json")
+        if os.path.exists(local_tfvars_file) and os.path.exists(local_config_file):
+            with open(local_tfvars_file) as tf:
+                local_tfvars = json.load(tf)
+            with open(local_config_file) as config:
+                local_config = json.load(config)
+                index = local_config["remote_state_bucket"]["resource_group_name"].find(
+                    "-"
+                )
+                local_config["prefix"] = local_config["remote_state_bucket"][
+                    "resource_group_name"
+                ][:index]
+            return bool(local_config["prefix"] != local_tfvars["prefix"])
+        else:
+            return False
+
+    def remove_matcha_dir(self) -> None:
+        """Removes the project's .matcha directory"."""
+        project_directory = os.getcwd()
+        target = os.path.join(project_directory, ".matcha")
+        if os.path.exists(target):
+            shutil.rmtree(target)
```

Comment:
I wonder if we could remove `os.getcwd()` (absolute file paths) and just use relative file paths. I'm not sure what the best practice for this is. Tagging @JonoCX for some clarification

Reply:
I think best practice is to use relative paths because the `.matcha` directory will always be relative

Reply:
There's a potential for issues to crop up when running tests at different levels, e.g., `pytest tests/test_runners`, as the relative path will be different because of the entry point when running the tests - the absolute paths avoid this. 

It might be worth testing the relative path approach to see what happens, but if the absolute paths work then let's go with that (it's the most straightforward)